routing.go

```diff
@@ -103,54 +107,33 @@ func (ps *proxyRouting) fetch(ctx context.Context, key string) (rb []byte, err e
 	}
 	defer resp.Body.Close()
 
-	// Read at most 10 KiB (max size of IPNS record).
-	rb, err = io.ReadAll(io.LimitReader(resp.Body, 10240))
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("routing/get RPC returned unexpected status: %s, body: %q", resp.Status, string(rb))
-	}
-
-	parts := bytes.Split(bytes.TrimSpace(rb), []byte("\n"))
-	var b64 string
-
-	for _, part := range parts {
-		var evt routing.QueryEvent
-		err = json.Unmarshal(part, &evt)
-		if err != nil {
-			return nil, fmt.Errorf("routing/get RPC response cannot be parsed: %w", err)
-		}
-
-		if evt.Type == routing.Value {
-			b64 = evt.Extra
-			break
-		}
-	}
-
-	if b64 == "" {
-		return nil, errors.New("routing/get RPC returned no value")
+		return nil, fmt.Errorf("unexpected status from remote gateway: %s", resp.Status)
 	}
 
-	rb, err = base64.StdEncoding.DecodeString(b64)
+	rb, err := io.ReadAll(io.LimitReader(resp.Body, int64(ipns.MaxRecordSize)))
 	if err != nil {
 		return nil, err
 	}
 
-	entry, err := ipns.UnmarshalRecord(rb)
+	rec, err := ipns.UnmarshalRecord(rb)
 	if err != nil {
 		return nil, err
 	}
 
-	err = ipns.ValidateWithName(entry, name)
+	err = ipns.ValidateWithName(rec, name)
 	if err != nil {
 		return nil, err
 	}
 
 	return rb, nil
 }
 
-func (ps *proxyRouting) getRandomKuboURL() string {
-	return ps.kuboRPC[ps.rand.Intn(len(ps.kuboRPC))]
+func (ps *proxyRouting) getRandomGatewayURL() string {
+	return ps.gatewayURLs[ps.rand.Intn(len(ps.gatewayURLs))]
```

Comment:
nit: let's trim trailing slash to avoid problems when someone passed gateway url with one

```suggestion
func (ps *proxyRouting) getRandomGatewayURL() string {
	return strings.TrimSuffix(ps.gatewayURLs[ps.rand.Intn(len(ps.gatewayURLs))], "/")
```